services/src/main/java/org/keycloak/crypto/MacSecretSignatureProvider.java

```diff
@@ -17,7 +17,9 @@
 package org.keycloak.crypto;
 
 import org.keycloak.common.VerificationException;
+import org.keycloak.jose.jwk.JWK;
 import org.keycloak.models.KeycloakSession;
+import org.keycloak.util.JWKSUtils;
 
 public class MacSecretSignatureProvider implements SignatureProvider {
```

Comment:
How about using `MacSecretClientSignatureVerifierProvider` that is used for keycloak to verify the data that a client creates and signs (e.g. request object, JWS client assertion for client authentication)?

`MacSecretSignatureProvider` is used for keycloak to sign and verify JWS (e.g. access token, ID token). DPoP Proof is the data that a client creates and signs and keycloak verify.

As for DPoP, I'm afraid that HMAC method might not be used...